### PR TITLE
DOC: Adds documentation of functions exposed in numpy namespace

### PR DESCRIPTION
**Pull request by kritisingh1**
In this PR, I will be documenting all the functions that required to be so in the issue #12385 . I will also be adding the functions which were discovered undocumented while working on this pull request and were not mentioned in the above issue. The checklist is as follows, I will keep adding more items as I discover more
- [x] compare_chararrays 
- [x] byte_bounds
- [x] get_include
- [ ] _get_indent // looks like an internal helper function, not needed to be public
- [x] disp
- [x] deprecate_with_doc
- [x] deprecate
- [x] who

xref #13214

### CODE REVIEW

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -3336,6 +3336,41 @@
     """)
 
 
+add_newdoc('numpy.lib.utils', 'byte_bounds',
```
Comment (eric-wieser) [inline numpy/core/_add_newdocs.py]: This isn't the right approach.

Functions defined in python should use regular docstrings - `_add_newdocs` is a hack to make adding docstrings to C function easier. It should not be used for python functions.

File: doc/source/reference/routines.other.rst [hunk 2]
```diff
@@ -35,3 +36,20 @@ NumPy version comparison
...
+   get_include
+   deprecate
+   deprecate_with_doc
+
+Matlab-like Functions
+---------------------
+.. autosummary::
+   :toctree: generated/
+
+   who
```
Comment (eric-wieser) [inline doc/source/reference/routines.other.rst]: I think you only need to touch the RST file here, there should be no need to duplicate the docstrings

File: numpy/core/_add_newdocs.py (hunk 1 above)
Comment (kritisingh1) [reply numpy/core/_add_newdocs.py]: @eric-wieser that means I need to only include docstrings of these functions? However, there are `_add_newdocs` used for functions like `may_share_memory` as well which are python functions defined in `multiarray.py`, could you explain why these two cases are different? Thanks.

Comment (eric-wieser) [reply numpy/core/_add_newdocs.py]: Good catch, those docstrings should also not be there. Fixed in gh-13876.

**Comment by mattip** [comment]
Thanks @kritisingh1

**Pull request merged**